### DISCUSSION TITLE
[help] specifying an output directory with `quarto_render()` works but also results in an error

### DISCUSSION THREAD

**Discussion by brndngrhm** [ropensci/tarchetypes#207] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I read through[ this discussion](https://github.com/ropensci/tarchetypes/discussions/107) but am experiencing a slightly different outcome. I am able to get the rendered html to the right output directory, but my `targets`  pipeline also errors.

I have my output directory defined in `_quarto.yml` like this:
```
project:
  output-dir: _quarto_output
```

(I tried changing it to match the recommendation [here,](https://github.com/ropensci/tarchetypes/discussions/107#discussioncomment-5385274) but that just put the html in the `R/reports` directory instead of `_quarto_output`, so i went back to my original `_quarto.yml`) 

when i run this target, it successfully creates an HTML document (although it creates nested folders in `_quarto_output` which wasnt expected but good enough) **but** i also i get the error below
```
  tar_quarto(
    name = explore_data,
    path = "R/reports/explore-data.qmd",
    quiet = FALSE,
    execute_params = list(
      data = combined_files,
      col_name_mapping = col_name_lookup_table
    )
  )
```

```
Output created: ..\..\_quarto_output\R\reports\explore-data.html

Error:
! targets::tar_make() error

── Debug target explore_data ───────────────────────────────────────────────────────────────────────────────────
    tar_meta(explore_data)$error
    tar_workspace(explore_data)

── General debugging ──────────────────────────────────────────────────────────────────────────────────────────────────
    • tar_errored()
    • tar_meta(fields = any_of("error"), complete_only = TRUE)
    • tar_workspace()
    • tar_workspaces()

── How to ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
    • Debug: [https://books.ropensci.org/targets/debugging.html](vscode-file://vscode-app/c:/Users/user_name/AppData/Local/Programs/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)
    • Help: [https://books.ropensci.org/targets/help.html](vscode-file://vscode-app/c:/Users/user_name/AppData/Local/Programs/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)

── Last error message ─────────────────────────────────────────────────────────────────────────────────────────────────
    _paths_ missing files: R/reports/explore-data.html

── Last error traceback ───────────────────────────────────────────────────────────────────────────────────────────────
    No traceback available.
```

**Comment by wlandau** [comment, marked as answer]
> I am able to get the rendered html to the right output directory

Almost, but like you mention later, the output file ends up in `_quarto_output/R/reports/explore-data.html` instead of `_quarto_output/explore-data.html`. You'll see this if you run `quarto render R/reports/explore-data.qmd` in the command line, so that part is a Quarto issue.

> i also i get the error below

`targets` needs to know the location of the output HTML file because it needs to watch the file for changes, and it's tricky when you have a project-level `output-dir` in `_quarto.yml` and a non-project-level `path = "R/reports/explore-data.qmd"`. 083ca895eb7aa2e4b225f745fb9f54fb3fb4bebf accounts for `output-dir` in this particular case. Even though the actual output file is not exactly where you want it, hopefully that helps.

I have never agreed with Quarto's policy around output file paths. When resolving relative paths, Quarto anchors off the parent directory of the source file, whereas most other tools anchor off the current working directory. Plus, there's the error "--output option cannot specify a relative or absolute path" whenever you try to explicitly set an output path (I forget where that Quarto issue thread lives).

**Reply by brndngrhm** [reply]
Thanks! i installed the latest dev version and the error is gone
